src/com/palmergames/bukkit/towny/regen/PlotBlockData.java

```diff
@@ -122,8 +122,8 @@ public boolean restoreNextBlock() {
 		while (reverse > 0) {
 			reverse--; //regen bottom up to stand a better chance of restoring tree's and plants.
 			y = height - (reverse % height);
-			x = (int) (reverse / height) % size;
-			z = ((int) (reverse / height) / size) % size;
+			x = (reverse / height) % size;
+			z = (reverse / height / size) % size;

```

Comment:
Wondering if this needs the ints to keep whole numbers. Have you tested the regeneration of blocks?

Reply:
This is fine since modulus cannot return anything but an integer. 
These statements are essentially
`(int / int) % int)`